Scenarios/Logging-and-Dependency-Injection/ReadMe.md

```diff
@@ -0,0 +1,49 @@
+
+### Simple Dependency Injection
+
+The `Service` class is registered as a singleton with DI as part of the process initialization. CoreWCF will look to see if the service class is registered with DI when it needs an instance. If none is registered, it will construct an instance of the class using reflection (provided there is a default constructor).
+
+This same DI mechanism can be used to pass any necessary utility classes into the service by adding them to the service constructor. 
+
+The logger is created for the `LoggingSample.Service` type, and is configured in [appsettings.json](LoggingSampple/appsettings.json) to log entries for Information level and higher. 
+
+### Scoped Dependency Injection
+
+The `Service2` class demonstrates how to inject a scoped dependency into the service class. The methods that need the dependency include additional parameters for those dependencies, which are marked with the `[Injected]` attribute. Provided that the class is defined as partial, CoreWCF will use code generation to create methods that will fulfill the service contract and supply the dependencies to the.
+
+For example, the app code:
+``` C#
+public CompositeType GetDataUsingDataContract(CompositeType composite, [Injected] ILogger<Service2> localLogger) 
+{
+    ...
+}
+```
+
+is supplemented by the following generated code:
+
+``` C#
+        public LoggingSample.CompositeType GetDataUsingDataContract(LoggingSample.CompositeType composite)
+        {
+            var serviceProvider = CoreWCF.OperationContext.Current.InstanceContext.Extensions.Find<IServiceProvider>();
+            if (serviceProvider == null) throw new InvalidOperationException("Missing IServiceProvider in InstanceContext extensions");
+            if (CoreWCF.OperationContext.Current.InstanceContext.IsSingleton)
+            {
+                using (var scope = serviceProvider.CreateScope())
+                {
+                    var d0 = scope.ServiceProvider.GetService<Microsoft.Extensions.Logging.ILogger<LoggingSample.CompositeType>>();
+                    return GetDataUsingDataContract(composite, d0);
+                }
+            }
+            var e0 = serviceProvider.GetService<Microsoft.Extensions.Logging.ILogger<LoggingSample.CompositeType>>();
+            return GetDataUsingDataContract(composite, e0);
+        }
+```
+ 
+This sample uses DI to inject a logger, but this pattern can be used for other dependencies such as a [DbContext](https://docs.microsoft.com/dotnet/api/microsoft.entityframeworkcore.dbcontext), where localized scoping is required.
+
```

Comment:
There are 2 details that need to be specified.  
1. This only works if you use DI for your service instance. It's an opt-in feature (as there's overhead with creation of the scope) and is enabled by using DI for your service instance.  
2. We should specify that the DI scope is matched to the InstanceContextMode. If they are using InstanceContextMode.Single, then all service calls will share a single DI scope. If InstanceContextMode is either PerCall or PerSession, then a new scope is created for every call or session. The default is PerSession, which is effectively PerCall whenever you aren't using a sessionful binding. This is important is the injected instance isn't thread safe, such as is the case with DbContext.